web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputerSetup/DocumentConnectMyComputerSetup.tsx

```diff
@@ -76,23 +81,115 @@ function Information(props: { onSetUpAgentClick(): void }) {
 }
 
 function AgentSetup() {
+  const ctx = useAppContext();
+
+  const [setUpRolesAttempt, runSetUpRolesAttempt] = useAsync(
+    useCallback(() => wait(1_000), [])
+  );
+  const [downloadAgentAttempt, runDownloadAgentAttempt] = useAsync(
+    useCallback(
+      () => ctx.connectMyComputerService.downloadAgent(),
+      [ctx.connectMyComputerService]
+    )
+  );
+  const [generateConfigFileAttempt, runGenerateConfigFileAttempt] = useAsync(
+    useCallback(() => wait(1_000), [])
+  );
+  const [joinClusterAttempt, runJoinClusterAttempt] = useAsync(
+    useCallback(() => wait(1_000), [])
+  );
+
   const steps = [
-    'Setting up roles',
-    'Downloading the agent',
-    'Generating the config file',
-    'Joining the cluster',
+    {
+      name: 'Setting up roles',
+      attempt: setUpRolesAttempt,
+    },
+    {
+      name: 'Downloading the agent',
+      attempt: downloadAgentAttempt,
+    },
+    {
+      name: 'Generating the config file',
+      attempt: generateConfigFileAttempt,
+    },
+    {
+      name: 'Joining the cluster',
+      attempt: joinClusterAttempt,
+    },
   ];
 
+  const runSteps = useCallback(async () => {
+    // uncomment when implemented
+    const actions = [
+      // runSetUpRolesAttempt,
+      runDownloadAgentAttempt,
+      runGenerateConfigFileAttempt,
+      // runJoinClusterAttempt,
+    ];
+    for (const action of actions) {
+      const [, error] = await action();
+      if (error) {
+        break;
+      }
+    }
+  }, [
+    runSetUpRolesAttempt,
+    runDownloadAgentAttempt,
+    runGenerateConfigFileAttempt,
+    runJoinClusterAttempt,
+  ]);
+
+  useEffect(() => {
+    runSteps();
+  }, [runSteps]);
```

Comment:
Run them only if all of them have not been touched yet, that is their status is equal to `''`.

Reply:
Done